src/CitationViewBuilder.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\stanford_publication;
+
+use Drupal\Core\Entity\EntityViewBuilder;
+use Drupal\Core\Render\Element;
+use Drupal\stanford_publication\Entity\CitationInterface;
+
+/**
+ * Class CitationViewBuilder.
+ *
+ * @package Drupal\stanford_publication
+ */
+class CitationViewBuilder extends EntityViewBuilder {
+
+  /**
+   * {@inheritDoc}
+   */
+  public function build(array $build) {
+    // This assumes the view mode name matches the constant name in the
+    // interface.
+    $constant = CitationInterface::class . '::' . strtoupper($build['#view_mode']);
+
+    if (defined($constant) && $style = constant($constant)) {
+      foreach (Element::children($build) as $child) {
+        unset($build[$child]);
+      }
+      $build['citation']['#markup'] = $build['#citation']->getBibliography($style);
+      return $build;
+    }
+
+    $build = parent::build($build);
+    return $this->buildDateDisplay($build);
+  }
+
+  /**
+   * Consolidate the year, month and day fields into a single string.
+   *
+   * @param array $build
+   *   Render array.
+   *
+   * @return array
+   *   Modified render array
+   */
+  protected function buildDateDisplay(array $build): array {
+    if (!isset($build['su_year'])) {
+      unset($build['su_month'], $build['su_day']);
+      return $build;
+    }
```

Comment:
If the year doesn't exist why do you remove day/month and return?

Reply:
Publications need a year if they have a month and/or day, so if no year is available, force the month and day to be removed.

I added that as a comment.